src/storage/blockchain/onchain-issuer.ts

```diff
@@ -0,0 +1,82 @@
+import { DID, Id, chainIDfromDID } from '@iden3/js-iden3-core';
+import { Contract, ethers } from 'ethers';
+import { INonMerklizedIssuerABI as abi } from '@iden3/onchain-non-merklized-issuer-base-abi';
+import { Options } from '@iden3/js-jsonld-merklization';
+import { W3CCredential } from '../../verifiable';
+import { OnchainNonMerklizedIssuerAdapter } from './onchain-issuer-adapter/non-merklized/version/v0.0.1/onchain-non-merklized-issuer-adapter';
+import { EthConnectionConfig } from '..';
+
+enum OnchainIssuerVersion {
+  'v0.0.1' = '0.0.1'
+}
+
+/**
+ * Represents an adapter for interacting with on-chain issuers.
+ *
+ * @public
+ * @beta
+ * @class OnchainIssuer
+ */
+export class OnchainIssuer {
```

Comment:
it must implemnt interface, so it will be possible to use stub or other way to get credential, e.g. if onchainIssuer will be used in handler it must be possible to replace with mock / other implementation and provide enough information